### PR TITLE
Feat/206 product details component

### LINKED ISSUE #206: [Feature]: [FEAT] User Dashboard Product Details Component
### Description

Develop a Product Details Component for the User Dashboard. This component displays detailed information about a selected product, allowing users to view and manage individual product details efficiently.

### Acceptance Criteria

- The component is implemented according to the provided design, displaying alongside the Product Lists Component.
- The layout is responsive and integrates seamlessly with the existing dashboard.

### Requirements:

- Product name as the header
- Close button (X) to dismiss the details view
- Product image
- Key product information: Product ID, Category, Date added, Stock, Price
- Product description
- Edit and Delete buttons

### Tests:

- Verify that the component displays correctly when a product is selected.
- Ensure all product details are accurately presented.
- Test the close button functionality.
- Check that the product image loads and displays properly.
- Verify the Edit and Delete buttons are present (functionality can be tested if implemented).

### Tasks:

- [x] - Develop the Product Details Component based on the design.
- [x] - Implement the header with product name and close button.
- [x] - Create an image container for the product photo.
- [x] - Add fields for Product ID, Category, Date added, Stock, and Price.
- [x] - Implement a section for the product description.
- [x] - Add Edit and Delete buttons at the bottom.
- [x] - Ensure the component can receive and display data for the selected product.
- [x] - Implement the close functionality to dismiss the details view.
- [x] - Add appropriate alt text for the product image.
- [ ] - Optimize the component for responsive design.
- [x] - Implement error handling for missing product data.
- [x] - Create smooth transitions for opening and closing the details view.
- [x] - Ensure text formatting and potential truncation for long descriptions.
- [x] - Prepare Edit and Delete functionalities for future implementation if not immediately active.

### Timeframe
**Deadline: Saturday @ 10am**

### Task(s) Linked to this feature

- [User Dashboard Product Details Page](https://github.com/hngprojects/hng_boilerplate_nextjs/issues/204)

### Links

| [`FIGMA LINK`](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=771-9815&t=ipWq1g5RzBq6LSrk-0)

### Images

image

### PR DESCRIPTION
**Pull request by GreyyDaze**
# Description
**Closes #206**

# Changes proposed

## What were you told to do?
[Feature]: [FEAT] User Dashboard Product Details Component

## What did you do?
- Developed the Product Details Component, including a header with the product name, a close button, and an image container.
- Added fields for Product ID, Category, Date added, Stock, and Price, along with a section for the product description.
- Implemented Edit and Delete buttons, ensuring functionality for displaying and dismissing product details.
- Optimized the component for responsive design, including error handling for missing data and smooth transitions for opening and closing views.
- Added appropriate alt text for the product image and ensured text formatting, with preparations for future Edit and Delete functionalities.

# Check List (Check all the applicable boxes)
- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos
image

**After Clicking Close Button**
image2

**Lint Screenshot**
lint

**Test ScreenShot**
test

**Figma Link**
[Figma Link](https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?t=x3fmKmy6xKc1VbHS-0)

### CODE REVIEW

File: public/product_image.jpg
Comment (skeby) [inline public/product_image.jpg:1]: put this image inside the images folder asap `public/images/{yourpagename}/product_image.jpg`
put the icons folder inside {yourpagename} folder `public/images/{yourpagename}/icons`

File: src/components/common/ProductDetailsCard/ProductDetailsCard.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  dateAdded = "N/A",
+  stock = "N/A",
+  price = "N/A",
+  description = "No description available",
+  onClose,
+  onEdit,
+  onDelete,
+}) => {
+  const handleEdit = () => {
+    if (productID === "N/A") {
```
Comment (skeby) [inline src/components/common/ProductDetailsCard/ProductDetailsCard.tsx:39]: Remove all console.logs

File: src/components/common/ProductDetailsCard/ProductDetailsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  const handleDelete = () => {
+    if (productID === "N/A") {
+      console.error("Product ID is missing. Cannot delete.");
+    } else {
+      onDelete(productID);
+    }
+  };
+
+  const handleClose = (event: React.MouseEvent) => {
+    event.stopPropagation(); // Prevent event bubbling
```
Comment (skeby) [inline src/components/common/ProductDetailsCard/ProductDetailsCard.tsx]: Remove all comments asap

**Review by skeby — DISMISSED** [review]

File: public/product_image.jpg
Comment (GreyyDaze) [reply public/product_image.jpg:1]: fix

File: src/components/common/ProductDetailsCard/ProductDetailsCard.tsx (hunk 1 above)
Comment (GreyyDaze) [reply src/components/common/ProductDetailsCard/ProductDetailsCard.tsx:39]: fix

File: src/components/common/ProductDetailsCard/ProductDetailsCard.tsx (hunk 2 above)
Comment (GreyyDaze) [reply src/components/common/ProductDetailsCard/ProductDetailsCard.tsx]: fix

File: public/images/productdetails/icons/close.svg
Comment (prudentbird) [inline public/images/productdetails/icons/close.svg:1]: Use lucide react

**Review by prudentbird — CHANGES_REQUESTED** [review]

File: public/images/productdetails/icons/close.svg
Comment (GreyyDaze) [reply public/images/productdetails/icons/close.svg:1]: fix

**Review by prudentbird — APPROVED** [review]

**Comment by prudentbird** [comment]
Fix your tests so I can merge

**Comment by GreyyDaze** [comment]
@PrudentBird Mentor, I have fixed test.

**Pull request merged**